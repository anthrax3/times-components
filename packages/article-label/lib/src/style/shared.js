Object.defineProperty(exports,"__esModule",{value:true});var _styleguide=require("@times-components/styleguide");var styles={title:{fontFamily:_styleguide.fonts.supporting,fontSize:_styleguide.fontSizes.cardMetaMobile,fontWeight:"400",letterSpacing:1.2,marginBottom:0}};exports.default=styles;