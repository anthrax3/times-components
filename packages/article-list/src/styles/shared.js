const sharedStyle = {
  ArticleListBody: {
    padding: 20
  }
};

export default sharedStyle;
