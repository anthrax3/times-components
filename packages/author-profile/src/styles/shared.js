import {
  colours,
  fonts,
  fontSizes,
  spacing
} from "@times-components/styleguide";

const styles = {
  paginationContainer: {
    alignItems: "stretch",
    flexDirection: "row",
    justifyContent: "center"
  },
  paginationSpacing: {
    flex: 1,
    maxWidth: 800
  },
  listingErrorContainer: {
    alignItems: "center",
    flex: 1,
    maxWidth: 548,
    paddingLeft: spacing(1),
    paddingRight: spacing(1)
  },
  listingErrorHeading: {
    alignSelf: "center",
    color: colours.functional.brandColour,
    fontFamily: fonts.headline,
    fontSize: fontSizes.leadHeadline,
    marginBottom: spacing(2),
    marginTop: spacing(4)
  },
  listingErrorMessage: {
    color: colours.functional.secondary,
    fontFamily: fonts.bodyRegular,
    fontSize: fontSizes.body,
    textAlign: "center"
  },
  listingErrorButtonContainer: {
    bottom: 0,
    position: "absolute",
    width: "100%"
  },
  itemSeparator: {
    backgroundColor: colours.functional.keyline,
    height: 1
  }
};

export default styles;
