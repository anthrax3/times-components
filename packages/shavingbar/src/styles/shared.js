import {spacing} from "@times-components/styleguide";

const sharedStyle = {
  body: {
    display: "flex",
    flexDirection: "row",
    borderColor: "rgb(219, 219, 219)",
    borderStyle: "solid",
    borderWidth: 1,
    borderLeftWidth: 1,
    borderRightWidth: 1,
    height: spacing(11),
    marginTop: spacing(10),
    marginBottom: spacing(4),
    justifyContent: "space-between",
    alignItems: "center"
  },
  group: {
    display: "flex",
    flexDirection: "row",
    height: spacing(4),
    alignItems: "center"
  },
  text: {
    marginRight: spacing(1)
  },
  bubble: {
    display: "flex",
    boxSizing: "border-box",
    justifyContent: "center",
    alignItems: "center",
    borderStyle: "solid",
    borderColor: "rgb(219, 219, 219)",
    borderWidth: 1,
    marginLeft: spacing(1),
    borderRadius: spacing(5),
    padding: spacing(2),
    height: spacing(8),
    width: spacing(8),
  }
};

export default sharedStyle;
