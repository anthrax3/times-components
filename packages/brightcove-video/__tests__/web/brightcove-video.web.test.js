import React from "react";
import renderer from "react-test-renderer";


import shared, { defaultVideoProps } from "../shared";
import IsPaidSubscriber from "../../src/is-paid-subscriber";
import BrightcoveVideo from "../../src/brightcove-video";

describe("BrightcoveVideo on web", () => {
  shared();


  const testSubscriberAndVideoPaidStatus = (
    subscriberIsPaid,
    videoIsPaidOnly
  ) => {
    const tree = renderer
      .create(
        <IsPaidSubscriber.Provider value={subscriberIsPaid}>
          <BrightcoveVideo {...defaultVideoProps} paidonly={videoIsPaidOnly} />
        </IsPaidSubscriber.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  };

  it("renders a paidonly video correctly for unpaid users", () => {
    testSubscriberAndVideoPaidStatus(false, true);
  });

  it("renders a paidonly video correctly for paid users", () => {
    testSubscriberAndVideoPaidStatus(true, true);
  });

  it("renders a non-paidonly video correctly for unpaid users", () => {
    testSubscriberAndVideoPaidStatus(false, false);
  });

  it("renders a non-paidonly video correctly for paid users", () => {
    testSubscriberAndVideoPaidStatus(true, false);
  });
});
