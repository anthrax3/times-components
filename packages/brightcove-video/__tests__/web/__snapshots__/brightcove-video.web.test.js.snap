// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrightcoveVideo on android renders a non-paidonly video correctly for paid users 1`] = `
<div
  style={
    Object {
      "height": 200,
      "width": 300,
    }
  }
>
  <video
    autoPlay={undefined}
    className="video-js"
    controls={true}
    data-account="[account id]"
    data-application-id={true}
    data-embed="default"
    data-player="[player id]"
    data-video-id="[video id]"
    id="[video id]-[account id]-3"
    poster="[poster uri]"
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
</div>
`;

exports[`BrightcoveVideo on android renders a non-paidonly video correctly for unpaid users 1`] = `
<div
  style={
    Object {
      "height": 200,
      "width": 300,
    }
  }
>
  <video
    autoPlay={undefined}
    className="video-js"
    controls={true}
    data-account="[account id]"
    data-application-id={true}
    data-embed="default"
    data-player="[player id]"
    data-video-id="[video id]"
    id="[video id]-[account id]-2"
    poster="[poster uri]"
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
</div>
`;

exports[`BrightcoveVideo on android renders a paidonly video correctly for paid users 1`] = `
<div
  style={
    Object {
      "height": 200,
      "width": 300,
    }
  }
>
  <video
    autoPlay={undefined}
    className="video-js"
    controls={true}
    data-account="[account id]"
    data-application-id={true}
    data-embed="default"
    data-player="[player id]"
    data-video-id="[video id]"
    id="[video id]-[account id]-1"
    poster="[poster uri]"
    style={
      Object {
        "height": 200,
        "width": 300,
      }
    }
  />
</div>
`;

exports[`BrightcoveVideo on android renders a paidonly video correctly for unpaid users 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  style={
    Object {
      "height": "200px",
      "width": "300px",
    }
  }
/>
`;

exports[`BrightcoveVideo on android renders a play icon correctly 1`] = `
<View>
  <Svg
    height={70}
    viewBox="0 0 100 100"
    width={70}
  >
    <Rect
      fill="rgba(0,0,0)"
      fillOpacity="0.4"
      height="100"
      stroke="rgb(255,255,255)"
      strokeWidth="8"
      width="100"
    />
    <Polygon
      fill="rgb(255,255,255)"
      points="30,20 70,50 30,80"
    />
  </Svg>
</View>
`;

exports[`BrightcoveVideo on android renders correctly 1`] = `<[object Object] />`;

exports[`BrightcoveVideo on android renders correctly without a poster image 1`] = `<[object Object] />`;
